File: aifupan_project/ReviewAnalysis/Asr/WordsMarkVo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using ReviewAnalysis.vo;

namespace ReviewAnalysis.Asr
{
    public class WordsMarkVo
    {
        /// <summary>
        /// 词语类型  0：敏感词 1：关键词
        /// </summary>
        [JsonProperty("wordsType")]
        public int WordsType { get; set; }

        /// <summary>
        /// 记录要标注的词语位置，比如[0, 2]则表示标注当前段落第0个、第2个词语
        /// </summary>
        [JsonProperty("recordNeedsWordList")]
        public List<RecordNeedsWordVo> RecordNeedsWordList { get; set; }

        /// <summary>
        /// 匹配中的限定词
        /// </summary>
        [JsonProperty("restrictWord")]
        public string RestrictWord { get; set; }
        /// <summary>
        /// 限定词范围
        /// </summary>
        [JsonProperty("restrictRange")]
        public int RestrictRange { get; set; }

        /// <summary>
        /// 词语类型
        /// </summary>
        [JsonProperty("wordsTypeStr")]
        public string WordsTypeStr { get; set; }

        /// <summary>
        /// 来源类型 0：系统 1：客户自定义
        /// </summary>
        [JsonProperty("resourceType")]
        public int ResourceType { get; set; }

        /// <summary>
        /// 来源类型
        /// </summary>
        [JsonProperty("resourceTypeStr")]
        public string ResourceTypeStr { get; set; }

        /// <summary>
        /// 细分类型
        /// 关键词 0：促单 1：互动 2：其他
        /// 敏感词 0：广告 1：品牌 2：国家 3：限制词 4：其他
        /// </summary>
        [JsonProperty("type")]
        public int Type { get; set; }

        /// <summary>
        /// 细分类型
        /// </summary>
        [JsonProperty("typeStr")]
        public string TypeStr { get; set; }

        /// <summary>
        /// 平台类型 0：全平台 1：抖音 2：快手 3：微视
        /// </summary>
        [JsonProperty("platformType")]
        public int PlatformType { get; set; }

        /// <summary>
        /// 平台类型
        /// </summary>
        [JsonProperty("platformTypeStr")]
        public string PlatformTypeStr { get; set; }

        /// <summary>
        /// 关键词名字
        /// </summary>
        [JsonProperty("name")]
        public string Name { get; set; }

        /// <summary>
        /// 敏感词等级 0：1级,封号  1：2级,严重警告  2：3级警告
        /// </summary>
        [JsonProperty("level")]
        public int? Level { get; set; }

        /// <summary>
        /// 敏感词等级
        /// </summary>
        [JsonProperty("levelStr")]
        public string LevelStr { get; set; }

        /// <summary>
        /// 出现次数
        /// </summary>
        [JsonProperty("countNum")]
        public int CountNum { get; set; }

        /// <summary>
        /// 描述
        /// </summary>
        [JsonProperty("remarks")]
        public string Remarks { get; set; }

        /// <summary>
        /// 行业id
        /// </summary>
        [JsonProperty("tradeId")]
        public string TradeId { get; set; }

        /// <summary>
        /// 行业
        /// </summary>
        [JsonProperty("tradeStr")]
        public string TradeStr { get; set; }

        /// <summary>
        /// 分组
        /// </summary>
        [JsonProperty("groupStr")]
        public string GroupStr { get; set; }
        /// <summary>
        /// 关键词类型id
        /// </summary>
        [JsonProperty("cruxTypeId")]
        public string CruxTypeId { get; set; }
        /// <summary>
        /// 关键词类型信息
        /// </summary>
        [JsonProperty("cruxTypeInfo")]
        public CruxTypeInfoVo CruxTypeInfo { get; set; }
        /// <summary>
        /// 词语在文中出现的总次数
        /// </summary>
        [JsonProperty("totalNum")]
        public int TotalNum { get; set; }
        /// <summary>
        /// 概览
        /// </summary>
        [JsonProperty("overView")]
        public string OverView { get; set; }
    }
}

File: aifupan_project/fupan-server/replay-generic/src/main/java/com/jiuyu/replay/generic/vo/words/WordsMarkVo.java
package com.jiuyu.replay.generic.vo.words;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.util.List;

@Data
@Schema(description = "关键词/敏感词")
public class WordsMarkVo {

    /**
     * 记录要标注的词语位置，比如[0, 2]则表示标注当前段落第0个、第2个词语（用来兼容1.8.3及以前版本）
     */
    @Schema(description = "记录要标注的词语位置，比如[0, 2]则表示标注当前段落第0个、第2个词语（用来兼容1.8.3及以前版本）")
    private List<Integer> recordNeedsWordNumList;

    /**
     * 记录要标注的词语位置，比如[0, 2]则表示标注当前段落第0个、第2个词语
     */
    @Schema(description = "记录要标注的词语位置，比如[0, 2]则表示标注当前段落第0个、第2个词语")
    private List<RecordNeedsWordVo> recordNeedsWordList;

    /**
     * 词语类型  0：敏感词 1：关键词 2：白名单(只有客户自定义有)
     */
    @Schema(description = "词语类型  0：敏感词 1：关键词 2：白名单(只有客户自定义有)")
    private Integer wordsType;
    /**
     * 词语类型
     */
    @Schema(description = "词语类型")
    private String wordsTypeStr;
    /**
     * 来源类型 0：系统 1：客户自定义
     */
    @Schema(description = "来源类型 0：系统 1：客户自定义")
    private Integer resourceType;
    /**
     * 来源类型
     */
    @Schema(description = "来源类型")
    private String resourceTypeStr;
    /**
     * 细分类型
     * 关键词 0：促单 1：互动 2：其他
     * 敏感词 0：广告 1：品牌 2：国家 3：限制词 4：其他
     */
    @Schema(description = "细分类型 关键词 0：促单 1：互动 2：其他   敏感词 0：广告 1：品牌 2：国家 3：限制词 4：其他")
    private Integer type;
    /**
     * 细分类型
     */
    @Schema(description = "细分类型")
    private String typeStr;
    /**
     * 平台类型 0：全平台 1：抖音 2：快手 3：视频号
     */
    @Schema(description = "平台类型 0：全平台 1：抖音 2：快手 3：视频号")
    private Integer platformType;
    /**
     * 平台类型
     */
    @Schema(description = "平台类型")
    private String platformTypeStr;
    /**
     * 关键词名字
     */
    @Schema(description = "关键词名字")
    private String name;
    /**
     * 敏感词等级 0：1级,封号  1：2级,严重警告  2：3级警告
     */
    @Schema(description = "敏感词等级 0：1级,封号  1：2级,严重警告  2：3级警告")
    private Integer level;
    /**
     * 敏感词等级
     */
    @Schema(description = "敏感词等级")
    private String levelStr;
    /**
     * 需要标注的次数
     */
    @Schema(description = "需要标注的次数")
    private Integer countNum;
    /**
     * 描述
     */
    @Schema(description = "描述")
    private String remarks;
    /**
     * 行业id，1表示全行业
     */
    @Schema(description = "行业id，1表示全行业")
    private Long tradeId;
    /**
     * 行业
     */
    @Schema(description = "行业")
    private String tradeStr;
    /**
     * 行业id数组
     */
    @Schema(description = "行业id数组")
    private String tradeIdArr;
    /**
     * 分组
     */
    @Schema(description = "分组")
    private String groupStr;
    /**
     * 关键词类型id
     */
    @Schema(description = "关键词类型id")
    private Long cruxTypeId;
    /**
     * 关键词类型信息
     */
    @Schema(description = "关键词类型信息")
    private CruxTypeInfoVo cruxTypeInfo;
    /**
     * 词语在文中出现的总次数
     */
    @Schema(description = "词语在文中出现的总次数")
    private Integer totalNum;
    /**
     * 概览
     */
    @Schema(description = "概览")
    private String overView;
}
